### DISCUSSION TITLE
Provide a way to specify a default format for when merging pull requests

### DISCUSSION THREAD

**Discussion by julienw** [community/community#27467] [Pull Requests] [answered]
It would be good to be able to specify a format when merging pull requests.
Indeed the defaults are:
* for merging: `Merge pull request #NNN from <branch name>` as title, with the pull request title in the commit log.
* for squashing: `<Pull request title> (#NNN)`

It would be good to be able to configure this. For example in our project we'd like this:
* for merging: `<Pull request title> (Merge PR #NNN)`
* for squashing: `<Pull request title> (PR #NNN)`

We do that manually and it's cumbersome.

Thanks for your consideration!

**Comment by nene** [comment]
For me the default Github merge commit titles provide little useful information. For example:

```
Merge pull request #322 from sql-formatter-org/issue/sql-formatter#321
```

In my experience branch names are often not very descriptive. They're often heavily abbreviated. Sometimes you also get pull requests from master branch. Sometimes you start a branch with one idea, and then during development you diverge from it, so that in the end the change has not much to do with the original branch name, but you stick with the original name out of laziness.

At the same time pull request titles usually describe very well what the change is about. And even when the PR title is bad, it's easy to change it through Github web UI, unlike branch names.

**Comment by willsmythe** [comment, marked as answer]
This is partially solved by the features introduced in August to control the default commit message format for merge commit and squash merges: https://github.blog/changelog/2022-08-23-new-options-for-controlling-the-default-commit-message-when-merging-a-pull-request/

**Reply by nene** [reply]
Awesome!
